src/release-specification.ts

```diff
@@ -301,6 +313,55 @@ export async function validateReleaseSpecification(
           });
         }
       }
+
+      // Check to compel users to release new versions of dependencies alongside their dependents
+      if (
+        pkg &&
+        versionSpecifierOrDirective &&
+        (hasProperty(IncrementableVersionParts, versionSpecifierOrDirective) ||
+          isValidSemver(versionSpecifierOrDirective))
+      ) {
+        const missingDependencies = Object.keys({
+          ...(pkg.unvalidatedManifest.dependencies || {}),
+          ...(pkg.unvalidatedManifest.peerDependencies || {}),
+        }).filter((dependency) => {
+          if (
+            !project.workspacePackages[dependency]?.hasChangesSinceLatestRelease
+          ) {
+            return false;
+          }
+
+          const dependencyVersionSpecifierOrDirective =
+            unvalidatedReleaseSpecification.packages[dependency];
+
+          return !dependencyVersionSpecifierOrDirective;
+        });
+
+        if (missingDependencies.length > 0) {
+          errors.push({
+            message: [
+              `The following packages, which are dependencies of the package '${packageName}' being released, are missing from the release spec.`,
```

Comment:
Since they may also be peer dependencies, what do you think about:

``` suggestion
              `The following packages, which are dependencies or peer dependencies of the package '${packageName}' being released, are missing from the release spec.`,
```